doc/features/mapper/queries/README.md

```diff
@@ -0,0 +1,286 @@
+```
+
+#### Get few selected fields of the objects
+
+Get only name and description of the videos
+
+```javascript
+const result = await videoMapper.find({ userId }, { fields: ['name', 'description' ]});
+```
+
+#### Get objects with a specific order
+
+Get all videos posted by a user
+
+```javascript
+const result = await videoMapper.find({ userId }, { orderBy: { 'addedDate': 'desc' }});
+```
+
+## Insert
+
+Use the `insert()` method on a `ModelMapper` instance to *upsert* a new object.
+
+When a model is mapped to multiple tables, it will insert a row in each table when all the primary keys are specified
+ grouped in a logged batch (either all or none of the insert operations will succeed).
+
+Additionally, `insert()` supports conditional clause for [lightweight transactions (CAS)][lwt] that allows to 
+insert only if the row doesn't exist. Please note that using IF conditions will incur a non negligible performance 
+cost on the server-side so this should be used sparingly.
+
+### Usage examples
+
+#### Insert a single object into one or more tables
+
+Insert a video
+
+```javascript
+await videoMapper.insert({ videoId, name, addedDate, userId, description });
+```
+
+#### Insert few selected fields of an object
+
+Insert only the id, the name and the added date, regardless of the other properties specified in the object.
+
+```javascript
+await videoMapper.insert(video, { fields: ['videoId', 'name', 'description'] });
+```
+
+#### Insert an object if it doesn't exit
+
+Insert a video when there isn't a video with the same id.
+
+```javascript
+await videoMapper.insert({ videoId, name, description }, { ifNotExists: true });
+```
+
+## Update
+
+Use the `update()` method on a `ModelMapper` instance to *upsert* a new object.
+
+When a model is mapped to multiple tables, it will update a row in each table when all the primary keys are specified
+ grouped in a logged batch (either all or none of the update operations will succeed).
+
+Additionally, `update()` supports conditional clause for [lightweight transactions (CAS)][lwt] that allows to 
+specify the condition that has to be met for the update to occur. Please note that using IF conditions will incur a 
+non negligible performance cost on the server-side so this should be used sparingly.
+
+### Usage examples
+
+#### Update a single object into one or more tables
+
+Update a video
+
+```javascript
+await videoMapper.update({ videoId, name, addedDate, userId, description });
+```
+
+#### Update few selected fields of an object
+
+Update only the name and the added date, regardless of the other properties specified in the object.
+
+```javascript
+await videoMapper.update(video, { fields: ['videoId', 'name', 'description'] });
+```
+
+#### Update an object using a [conditional statement][lwt]
+
+Update a video when the existing name contains a certain value.
+
+```javascript
+await videoMapper.update({ videoId, name, description }, { when: { name: 'original name' } });
```

Comment:
`when` is translated to `IF` on `UPDATE` and `DELETE` queries, in this case it would be:

```javascript
videoMapper.update({ videoId, name, description }, { when: { name: 'original name' } })
```

gets translated to 

```cql
UPDATE videos SET name = ?, description = ? WHERE videoid = ? IF name = ?
```

Reply:
I misunderstood that.  `when` definitely makes more sense in this context then :+1: